custom_components/hacs/repositories/plugin.py

```diff
@@ -124,11 +124,13 @@ def update_filenames(self) -> None:
                         self.content.path.remote = "release"
                         return
 
-        for location in ("",) if self.repository_manifest.content_in_root else ("dist", ""):
-            for filename in valid_filenames:
-                if f"{location+'/' if location else ''}{filename}" in [
-                    x.full_path for x in self.tree
-                ]:
-                    self.data.file_name = filename.split("/")[-1]
-                    self.content.path.remote = location
-                    break
+        all_paths = tuple(x.full_path for x in self.tree)
```

Comment:
```suggestion
        all_paths = {x.full_path for x in self.tree}
```

Would this make sense to use a `set` here so it doesn't have to do a linear search of the the tuple for the `in`/contains?

Reply:
Its just a collection of strings, so it does not really matter how its stored